fsql/deser.py

```diff
@@ -87,31 +91,51 @@ def read_and_concat(
 
 
 class PandasReader(DataReader):
+    """Wraps various pandas read methods (parquet, json, csv, excel) into a single interface.
+    Behaviour can be customised via passing any kwargs to the constructor.
+    """
+
+    def __init__(self, input_format=InputFormat.AUTO, **pdread_kwargs):
```

Comment:
Do you think we will extend the signature in the future? I would rather not, but it's one of the reason why I do not like this kind of kwargs propagation that might be a bit dangerous sometimes... (although we use it quite often 😃 )

Reply:
I completely agree and I in general not use kwargs myself. But as the evil down there is already in place, what can I do :shrug: 

Idea I toyed with and which may end up to the next version is to require the user to provide directly the `read_single` function, thus leaving the user fully in control of the kwargs passing. I did not use it now as it would require user do what feels like boilerplate. So once we'll get to a situation where the signature needs to be extended too much, we may escape via this way, possibly keeping some of the previous functionality via some utils or factories or builders

I'm not really worried by it because the core thing that must stay is the DataReader interface which is not tainted by this at all

Reply:
I was thinking about a case when the signature extension might introduce some breaking changes, especially because of used kwargs. But let's not complicate it atm.